packages/ui5-application-writer/src/data/defaults.ts

```diff
@@ -53,6 +53,26 @@ export function mergeApp(app: App): App {
     );
 }
 
+/**
+ * Merges the provided ReadMe configuration with default values to be used by read me templates.
+ *
+ * @param {ReadMe} readMe - The configuration object to merge with defaults.
+ * @returns {ReadMe} The merged configuration object with defaults applied.
+ */
+export function mergeWithReadMeDefaults(readMe: ReadMe): ReadMe {
```

Comment:
Is this required outside of Fiori Elements generator?

Reply:
Yes, we should add default values if they are not provided through the App interface. While all mandatory elements will be accessible, the README.md template requires specific parameters such as `metadataFilename`, `serviceURL` etc. If `metadataFilename` is not defaulted when provided, the template will generate an error message as follows: 

```
console.error
    Error copying README template: ReferenceError: ****/templates/core/README.md:9
        7| |**Generation Platform**<br><%= genPlatform %>|
        8| |**Template Used**<br><%= templateLabel %>|
     >> 9| |**Service Type**<br><%= dataSourceLabel %>|<% if(metadataFilename) { %>
        10| |**Metadata File**<br><%= metadataFilename %>|<% } else { %>
        11| |**Service URL**<br><%= serviceUrl %>|<%}%>
        12| |**Module Name**<br><%= projectName %>|
    
    metadataFilename is not defined
```

Setting defaults helps prevent errors alike these template failures or crashes, which could occur if mandatory parameters are missing or incorrectly specified.

Reply:
The template should be updated in that case. The optional and required properties should be defined by the interface rather than the template implementation.

Reply:
oh I see we can use locals - will change the template  - thanks

Reply:
made use of locals